### PR TITLE
Add prometheus metrics (fixes #62, fixes #23)

### LINKED ISSUE #23: Add StatsD (or Prometheus) metrics
- Count accepted / rejected BZ events
- Measure API calls latency
- Measure execution time

Configure client from `STATSD_DSN` env var. Skip if empty.

### LINKED ISSUE #62: Usage metrics for JBI
Context:  Feedback from the ITEO program review 

* metrics for each configured action about how many syncs it sent from BZ to Jira 
* can go to grafana
* gives us an idea of how much manual work JBI has saved / automated

### PR DESCRIPTION
**Pull request by leplatrem**
Fixes #62 
Fixes #23

### CODE REVIEW

File: src/jbi/services.py [hunk 1]
```diff
@@ -13,17 +14,45 @@
...
+    the specieds methods are called.
+    """
+
+    counters: Dict[str, Counter] = {}
+
+    def __init__(self, wrapped, methods):
+        self.wrapped = wrapped
+        for method in methods:
+            if method not in self.counters:
+                self.counters[method] = Counter(method, f"Call of {method}()")
```
Comment (leplatrem) [inline src/jbi/services.py:28]: maybe there should be a prefix string here (eg. `prefix="jira."`)

Comment (bsieber-mozilla) [reply src/jbi/services.py:28]: or have `prefix` be a param; so this could be reused for the other services?

**Review by bsieber-mozilla — APPROVED** [review]
LGTM! thanks!

**Comment by bsieber-mozilla** [comment]
Where will we be able to view these metrics?

**Comment by leplatrem** [comment]
> Where will we be able to view these metrics?
>
I was going to ask you. Since you don't know, I guess that it was never setup. I assume you did since there were mentions of Prometheus in the Gunicorn conf. 

That means I need to make sure the Promotheus client is actually sending these metrics before merging this PR

**Comment by bsieber-mozilla** [comment]
> > Where will we be able to view these metrics?
> 
> I was going to ask you. Since you don't know, I guess that it was never setup. I assume you did since there were mentions of Prometheus in the Gunicorn conf.
> 
> That means I need to make sure the Promotheus client is actually sending these metrics before merging this PR

I believe the metrics should be sent here: https://earthangel-b40313e5.influxcloud.net/d/YS2KrS_nk/jbi?orgId=1

Are you looking to make charts and/or notification events from these metrics?

**Comment by leplatrem** [comment]
Existing metrics are reported here:  https://earthangel-b40313e5.influxcloud.net/d/YS2KrS_nk/jbi?orgId=1
Let's see if they show up once this gets deployed

**Pull request merged**